include/nuttx/spinlock.h

```diff
@@ -531,7 +531,13 @@ irqstate_t spin_lock_irqsave_wo_note(FAR volatile spinlock_t *lock)
 
   if (NULL == lock)
     {
-      int me = this_cpu();
+      int me;
+
+#ifdef CONFIG_SMP
+      me = up_cpu_index();
```

Comment:
define the up cupu index to zero when not in SMP mode

Reply:
up_cpu_index may not zero when in AMP mode

Reply:
why not revert https://github.com/apache/nuttx/pull/14161? or even https://github.com/apache/nuttx/pull/14190 since https://github.com/apache/nuttx/pull/14200 fix the dead lock.